### PR TITLE
adding confirmOverride modal

### PR DESCRIPTION
**Pull request by sharad-sharma**
Here I am using 3 atoms, one for storing error.credentials and 2 for handling modals. Idk any good way to store callback function to atom and to be able to call that on modal so, I have used these 3 atoms to do this task.

### CODE REVIEW

File: src/atoms/firebaseUserAtoms.ts [hunk 1]
```diff
@@ -43,6 +43,34 @@ firebaseUserAtom.onMount = setAtom => {
...
+  toJSON(): Object
+}>(
+  {
+    providerId: "google.com",
+    signInMethod: "google.com",
+    oauthIdToken: "absolutely-dummy-string",
+    oauthAccessToken: "absolutely-dummy-string",
+    toJSON: () => JSON.stringify(this)
+}
+);
```
Comment (thecodingwizard) [inline src/atoms/firebaseUserAtoms.ts:61]: I don't think we need this atom. It's only used in one location

File: src/atoms/firebaseUserAtoms.ts [hunk 2]
```diff
@@ -43,6 +43,34 @@ firebaseUserAtom.onMount = setAtom => {
...
+export const respondConfirmOverrideAtom = atom(
+  get => get(showConfirmModal),
+  (get, set, dataOverrideResponse: boolean) => {
+    if(dataOverrideResponse) {
+      // get(CredentialforDataOverrideAtom) is already been updated.
+      firebase.auth().signInWithCredential(get(CredentialforDataOverrideAtom));
+    }
+    set(showConfirmModal, false);
+  }
+)
```
Comment (thecodingwizard) [inline src/atoms/firebaseUserAtoms.ts:72]: These two should be combined into one atom. Also, the logic of signing in with credential should go in the `signInWithGoogle` atom, not in this atom. The show confirm modal atom should just accept a callback function; the logic for signing in should be in this callback function that's set by the `signInWithGoogle` atom

File: src/atoms/firebaseUserAtoms.ts [hunk 3]
```diff
@@ -79,9 +108,8 @@ export const signInWithGoogleAtom = atom(null, (get, set, _) => {
       .catch(error => {
         if (error.code === 'auth/credential-already-in-use') {
           // user already has an account. Sign in to that account and override our data.
-          confirmOverrideData(() => {
-            firebase.auth().signInWithCredential(error.credential);
-          });
+          set(CredentialforDataOverrideAtom, error.credential);
+          set(showConfirmModal, true);
```
Comment (thecodingwizard) [inline src/atoms/firebaseUserAtoms.ts:112]: Just set a singular atom here, passing in a callback for what happens when the user confirms data override. ie:

```
firebase.auth().signInWithCredential(error.credential);
```

File: src/components/ConfirmOverrideModal.tsx [hunk 4]
```diff
@@ -0,0 +1,72 @@
...
+                </div>
+              </div>
+            </div>
+          </div>
+          <div className="bg-gray-50 px-4 py-3 sm:px-6 sm:flex sm:flex-row-reverse">
+            <button
+              type="button"
+              className="w-full inline-flex justify-center rounded-md border border-transparent shadow-sm px-4 py-2 bg-red-600 text-base font-medium text-white hover:bg-red-700 focus:outline-none focus:ring-2 focus:ring-offset-2 focus:ring-red-500 sm:ml-3 sm:w-auto sm:text-sm"
+              onClick={() => {
+                setConfirmOverrideAtom(true);
```
Comment (thecodingwizard) [inline src/components/ConfirmOverrideModal.tsx:45]: call the callback function, then set the atom to nul

File: src/components/ConfirmOverrideModal.tsx [hunk 5]
```diff
@@ -0,0 +1,72 @@
...
+              }}
+            >
+              Continue
+            </button>
+            <button
+              type="button"
+              className="mt-3 w-full inline-flex justify-center rounded-md border border-gray-300 shadow-sm px-4 py-2 bg-white text-base font-medium text-gray-700 hover:bg-gray-50 focus:outline-none focus:ring-2 focus:ring-offset-2 focus:ring-indigo-500 sm:mt-0 sm:ml-3 sm:w-auto sm:text-sm"
+              onClick={() => {
+                setConfirmOverrideAtom(false);
```
Comment (thecodingwizard) [inline src/components/ConfirmOverrideModal.tsx:54]: set the atom to null without calling the callback function

**Review by thecodingwizard — CHANGES_REQUESTED** [review]
The code still uses three atoms. I've left some comments detailing how to reduce it down to one atom. The logic for signing in should go in the `signInWithGoogleAtom`. Let me know if anything is unclear :)

**Comment by thecodingwizard** [comment]
@sharad-sharma any updates? :)

**Comment by sharad-sharma** [comment]
@thecodingwizard 

I'm working on it. I will fix it soon.

**Pull request merged**